lib/update/updater.js

```diff
@@ -28,8 +33,16 @@ Updater.factory = function factory (config) {
 }
 
 Updater.prototype._fetchDeviceId = function _fetchDeviceId () {
-  return fs.readFileSync('/sys/class/net/wlan0/address',
-    {encoding: 'utf8'}).trim().replace(/:/g, '-')
+  const netInterfaces = os.networkInterfaces()
+  const firstWlanName = Object.keys(netInterfaces).filter(name => name.startsWith('wlan')).sort()[0]
+  const firstEthName = Object.keys(netInterfaces).filter(name => name.startsWith('eth')).sort()[0]
+  const mainInterfaceName = firstWlanName ? firstWlanName : firstEthName
+
+  if (mainInterfaceName) {
+    const macAddr = netInterfaces[mainInterfaceName][0].mac
+    return macAddr.replace(/:/g, '-')
+  }
+  else return null
```

Comment:
Consider just filtering ``netInterfaces`` for all ``mac`` that's not ``00:00:00:00:00:00``, sorting, and choosing the first one.

Reply:
Thought a little bit about your suggestion. Sounds better than the current one. I also thought that we don't need to sort the result, but we do since there might not be any guarantees that the order in which they're reported is maintained across reboots.
Going for it!